nixos/modules/tasks/filesystems/bcachefs.nix

```
@@ -93,19 +93,27 @@ let
     let
       mountUnit = "${utils.escapeSystemdPath (prefix + (lib.removeSuffix "/" fs.mountPoint))}.mount";
       device = firstDevice fs;
-      deviceUnit = "${utils.escapeSystemdPath device}.device";
+      mkDeviceUnit = device: "${utils.escapeSystemdPath device}.device";
+      deviceUnit = mkDeviceUnit device;
+      extractProperty =
+        prop: options: (map (lib.removePrefix "${prop}=") (builtins.filter (lib.hasPrefix prop) options));
+      normalizeUnits = unit: if lib.hasPrefix "/" unit then mkDeviceUnit unit else unit;
```

It needs to handle `/dev/` paths and other paths differently.

```suggestion
      mkMountUnit = path: "${utils.escapeSystemdPath path}.mount";
      normalizeUnits =
        unit:
        if lib.hasPrefix "/dev/" unit then
          mkDeviceUnit unit
        else if lib.hasPrefix "/" unit then
          mkMountUnit unit
        else
          unit;
```

---

oh it also supports mount paths?